src/app/options.cpp

```diff
@@ -0,0 +1,394 @@
+}
+
+QBtCommandLineParameters::QBtCommandLineParameters(const QProcessEnvironment &env)
+    : showHelp(false)
+#ifndef Q_OS_WIN
+    , showVersion(false)
+#endif
+#ifndef DISABLE_GUI
+    , noSplash(NO_SPLASH_OPTION.value(env))
+#else
+    , shouldDaemonize(DAEMON_OPTION.value(env))
+#endif
+    , webUiPort(WEBUI_PORT_OPTION.value(env, -1))
+    , profileDir(PROFILE_OPTION.value(env))
+    , portableMode(PORTABLE_OPTION.value(env))
+    , configurationName(CONFIGURATION_OPTION.value(env))
+{
+}
+
+QBtCommandLineParameters parseCommandLine(const QStringList &args)
+{
+    QBtCommandLineParameters result {QProcessEnvironment::systemEnvironment()};
+
+    for (int i = 1; i < args.count(); ++i) {
+        const QString &arg = args[i];
+
+        if ((arg.startsWith("--") && !arg.endsWith(".torrent")) ||
+            (arg.startsWith("-") && (arg.size() == 2))) {
+            // Parse known parameters
+            if ((arg == SHOW_HELP_OPTION)) {
+                result.showHelp = true;
+            }
+#ifndef Q_OS_WIN
+            else if (arg == SHOW_VERSION_OPTION) {
+                result.showVersion = true;
+            }
+#endif
+            else if (arg == WEBUI_PORT_OPTION) {
+                result.webUiPort = WEBUI_PORT_OPTION.value(arg);
+                if (( result.webUiPort < 1) || ( result.webUiPort > 65535) )
+                    throw CommandLineParameterError(QObject::tr("%1 must specify the correct port (1 to 65535).")
+                                                    .arg(QLatin1String("--webui-port")));
+            }
+#ifndef DISABLE_GUI
+            else if (arg == NO_SPLASH_OPTION) {
+                result.noSplash = true;
+            }
+#else
+            else if (arg == DAEMON_OPTION) {
+                result.shouldDaemonize = true;
+            }
+#endif
+            else if (arg == PROFILE_OPTION) {
+                result.profileDir = PROFILE_OPTION.value(arg);
+            }
+            else if (arg == PORTABLE_OPTION) {
+                result.portableMode = true;
+            }
+            else if (arg == CONFIGURATION_OPTION) {
+                result.configurationName = CONFIGURATION_OPTION.value(arg);
+            }
+            else {
+                // Unknown argument
+                result.unknownParameter = arg;
+                break;
+            }
+        }
+        else {
+            QFileInfo torrentPath;
+            torrentPath.setFile(arg);
+
+            if (torrentPath.exists())
+                result.torrents += torrentPath.absoluteFilePath();
+            else
+                result.torrents += arg;
+        }
+    }
+
+    return result;
+}
+
+CommandLineParameterError::CommandLineParameterError(const QString &messageForUser)
+    : std::runtime_error(messageForUser.toLocal8Bit().data())
+    , m_messageForUser(messageForUser)
+{
+}
+
+const QString& CommandLineParameterError::messageForUser() const
+{
+    return m_messageForUser;
+}
+
+QString makeUsage(const QString &prg_name)
```

Comment:
Please fix param name (use camelCase).
I understand that most likely it got you inherited, but I hope it's not too much trouble you.